Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -766,6 +766,9 @@ Device::Device(Gfx::IntSize size)
 {
     m_options.scissor_box = m_frame_buffer->rect();
     m_options.viewport = m_frame_buffer->rect();
+
+    // Ensure we can always append 3 vertices unchecked
+    m_clipped_vertices.ensure_capacity(3);
```

Comment:
Seems like candidate for inline capacity

Reply:
Wouldn't that generate an additional condition for every append? I.e. see `Vector::data()`

Reply:
For reference, this would not be an improvement since the inline buffer would be skipped entirely as soon as the outline buffer is allocated, effectively resulting in the same behavior as the code in this PR but with an additional conditional.